src/overlays/actors/ovl_Bg_Gjyo_Bridge/z_bg_gjyo_bridge.c

```diff
@@ -21,17 +29,94 @@ const ActorInit Bg_Gjyo_Bridge_InitVars = {
     (ActorFunc)BgGjyoBridge_Update,
     (ActorFunc)BgGjyoBridge_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Gjyo_Bridge/BgGjyoBridge_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Gjyo_Bridge/BgGjyoBridge_Destroy.s")
+InitChainEntry initChain[] = { ICHAIN_F32(unk_F8, 800, ICHAIN_CONTINUE),
+                               ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP) };
+
+extern u32 D_06000600;
+extern u32 D_06000DB8;
+extern u32 D_02002640;
+
+void BgGjyoBridge_Init(BgGjyoBridge* this, GlobalContext* globalCtx) {
+    s32 pad;
+    DynaCollisionContext* dynaCollisionContext;
+    s32 local_c;
+
+    local_c = 0;
+
+    Actor_ProcessInitChain(&this->dyna, &initChain);
+    DynaPolyInfo_SetActorMove(this, 0);
+    DynaPolyInfo_Alloc(&D_06000DB8, &local_c);
+
+    dynaCollisionContext = &globalCtx->colCtx.dyna;
+
+    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, dynaCollisionContext, this, local_c);
+
+    if (gSaveContext.event_chk_inf[4] & 0x2000) {
+        this->actionFunc = &func_808787A4;
+    } else {
+        this->dyna.actor.draw = 0;
+        func_8003EBF8(globalCtx, dynaCollisionContext, this->dyna.dynaPolyId);
+        this->actionFunc = &func_808787B0;
+    }
+}
+
+void BgGjyoBridge_Destroy(BgGjyoBridge* this, GlobalContext* globalCtx) {
+    DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+}
+
+void func_808787A4(BgGjyoBridge* this, GlobalContext* globalCtx) {
+}
+
+void func_808787B0(BgGjyoBridge* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if ((gBitFlags[3] & gSaveContext.quest_items) && (gBitFlags[4] & gSaveContext.quest_items) &&
+        (INV_CONTENT(ITEM_ARROW_LIGHT) == ITEM_ARROW_LIGHT) && (player->actor.posRot.pos.x > -70.0f) &&
+        (player->actor.posRot.pos.x < 300.0f) && (player->actor.posRot.pos.y > 1340.0f) &&
+        (player->actor.posRot.pos.z > 1340.0f) && (player->actor.posRot.pos.z < 1662.0f) &&
+        (func_800BFC84(globalCtx) == 0)) {
+        globalCtx->csCtx.segment = SEGMENTED_TO_VIRTUAL(&D_02002640);
+        gSaveContext.cutscene_trigger = 1;
+        this->actionFunc = &func_80878904;
+    }
+}
+
+void func_80878904(BgGjyoBridge* this, GlobalContext* globalCtx) {
```

Comment:
and BgGjyoBridge_SpawnBridge for this. spawn may not be the best word, but something along those lines. no worries if you dont want to name stuff now, but figured its pretty easy to tell what its doing